op-batcher/batcher/driver.go

```diff
@@ -324,6 +331,66 @@ func (l *BatchSubmitter) loop() {
 	}
 }
 
+// checkRecentTxsOnStart Check to see if there was a batcher tx sent recently that
+// still needs more block confirmations before being considered finalized
+func (l *BatchSubmitter) checkRecentTxsOnStart() {
+	currBlock, err := l.Txmgr.BlockNumber(l.shutdownCtx)
+	if err != nil {
+		l.Log.Error("failed to retrieve current block number", "err", err)
+		return
+	}
+
+	currentBlock := new(big.Int).SetUint64(currBlock)
+	currentNonce, err := l.L1Client.NonceAt(l.shutdownCtx, l.Txmgr.From(), currentBlock)
+	if err != nil {
+		l.Log.Error("Failed to retrieve current nonce", "err", err)
+		return
+	}
+
+	rollupClient, err := l.EndpointProvider.RollupClient(l.shutdownCtx)
+	if err != nil {
+		l.Log.Error("Failed to get rollup client", "err", err)
+		return
+	}
+	rollupConfig, err := rollupClient.RollupConfig(l.shutdownCtx)
+	if err != nil {
+		l.Log.Error("Failed to retrieve rollup config", "err", err)
+		return
+	}
+	blockConfirms := rollupConfig.VerifierConfDepth
+
+	previousBlockBig := new(big.Int)
+	previousBlockBig.Sub(currentBlock, big.NewInt(int64(blockConfirms)))
+	previousNonce, err := l.L1Client.NonceAt(l.shutdownCtx, l.Txmgr.From(), previousBlockBig)
+	if err != nil {
+		l.Log.Error("Failed to retrieve previous nonce", "err", err)
+		return
+	}
+
+	if currentNonce == previousNonce {
+		l.Log.Info("No recent batcher txs detected. No need to wait for rollup sync")
+		return
+	}
```

Comment:
we would still need to wait for node-sync on the oldest block, so not return here, but set the `batcherTxBlock` to the `oldestBlock`. Note that if we move this into its own function `CheckRecentTxs`, we `return oldestBlock, false, nil` here.